File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    let answer = 0;

    dfs(numbers[0], 1);
    dfs(-numbers[0], 1);

    function dfs(x, y) {
        if (y === numbers.length) {
            if (x === target) {
                answer++;
            }
            return;
        }

        dfs(x + numbers[y], y + 1);
        dfs(x - numbers[y], y + 1);
    }

    return answer;
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    cnt = 0

    def dfs(n, i):
        nonlocal cnt

        if i == len(numbers):
            if n == target:
                cnt += 1
            return

        dfs(n + numbers[i], i + 1)
        dfs(n - numbers[i], i + 1)

    dfs(0, 0)

    return cnt